src/main/java/randoop/condition/BooleanExpression.java

```diff
@@ -0,0 +1,312 @@
+ */
+public class BooleanExpression {
+
+  /** The basename for the expression class name. It is used for compiling the method. */
+  private static final String EXPRESSION_CLASS_NAME = "RandoopExpressionClass";
+
+  /** The name generator to use to generate class names. */
+  private static final NameGenerator nameGenerator = new NameGenerator(EXPRESSION_CLASS_NAME);
+
+  /** The {@code java.lang.reflect.Method} to test this expression */
+  private final Method expressionMethod;
+
+  /** The comment describing this expression */
+  private final String comment;
+
+  /**
+   * The Java source code in {@link ObjectContract} format. See {@link #getContractSource()} for
+   * format details.
+   */
+  private final String contractSource;
+
+  /**
+   * Creates a {@link BooleanExpression} that calls the method to evaluate the expression.
+   *
+   * @param expressionMethod the reflection {@code Method} for the expression
+   * @param comment a comment describing this expression
+   * @param contractSource the source code for this expression (see {@link #getContractSource()} for
+   *     format details)
+   */
+  BooleanExpression(Method expressionMethod, String comment, String contractSource) {
+    this.expressionMethod = expressionMethod;
+    this.comment = comment;
+    this.contractSource = contractSource;
+  }
+
+  /**
+   * Creates a {@link BooleanExpression} for evaluating an expression (see {@link
+   * randoop.condition.specification.Guard}) of a specification.
+   *
+   * @param signature the signature for the expression method to be created. The class name of the
+   *     expression method signature is ignored and a new name is generated using {@link
+   *     #nameGenerator}.
+   * @param declarations the parameter declaration string for the expression method to be created,
+   *     including parameter names and wrapped in parentheses
+   * @param expressionSource the source code for a Java expression to be used as the body of the
+   *     expression method
+   * @param contractSource a Java expression that is the source code for the expression, in the
+   *     format of {@link BooleanExpression#getContractSource()}
+   * @param comment the comment describing the expression
+   * @param compiler the compiler to used to compile the expression method
+   * @return the {@link BooleanExpression} that evaluates the given expression source on parameters
+   *     described by the declaration string
+   */
+  static BooleanExpression createBooleanExpression(
+      RawSignature signature,
+      String declarations,
+      String expressionSource,
+      String contractSource,
+      String comment,
+      SequenceCompiler compiler) {
+    Method expressionMethod = createMethod(signature, declarations, expressionSource, compiler);
+    return new BooleanExpression(expressionMethod, comment, contractSource);
+  }
+
+  /**
+   * Returns the {@link BooleanExpression} that checks the expression with the given argument values
```

Comment:
I don't understand the semantics and intended use of this method.  Can you write a more complete specification?

As one example of an ambiguity, are the given argument values a subset of all the arguments?  Or must the size of `args` be the same as the current size of the arguments?